agent/proxycfg/snapshot.go

```diff
@@ -7,24 +7,39 @@ import (
 	"github.com/mitchellh/copystructure"
 )
 
-type configSnapshotConnectProxy struct {
-	Leaf                     *structs.IssuedCert
-	DiscoveryChain           map[string]*structs.CompiledDiscoveryChain // this is keyed by the Upstream.Identifier(), not the chain name
-	WatchedUpstreams         map[string]map[string]context.CancelFunc
+// A shared data structure that contains information about discovered upstreams
+type ConfigSnapshotUpstreams struct {
+	Leaf *structs.IssuedCert
```

Comment:
The reason it is here is because we use it for both connect proxy and ingress gateways, specifically the `handleUpdateUpstreams` function that we have to reuse code between the two implementations.

Reply:
Maybe leave yourself a TODO for later to come back and see if there's a better term to describe this bucket of settings.

Reply:
Added a TODO item for us to come and think of a better name. 👍